backend/tournesol/views/comparison.py

```diff
@@ -132,9 +132,17 @@ def perform_create(self, serializer):
         comparison.entity_2.auto_remove_from_rate_later(
             poll=poll, user=self.request.user
         )
-
-        if settings.UPDATE_MEHESTAN_SCORES_ON_COMPARISON and poll.algorithm == ALGORITHM_MEHESTAN:
-            update_user_scores(poll, user=self.request.user)
+        if (
+            settings.UPDATE_MEHESTAN_SCORES_ON_COMPARISON
+            and poll.algorithm == ALGORITHM_MEHESTAN
+        ):
+            update_user_scores(
+                poll,
+                user=self.request.user,
+                uid_a=self.request.data["entity_a"]["uid"],
+                uid_b=self.request.data["entity_b"]["uid"],
+                delete_comparison_case=False,
+            )
```

Comment:
What's happening if the main mehestan job is running while a new comparison will be submitted? Or 2 concurrent comparisons are submitted about the same entities ?
Is there not a risk that these updates may conflict ? Do we need some lock mechanism ?

Reply:
Good question about entity conflict/lock. 


For the main mehestan job, we should disable the heuristic in that windows frame, I guess it is a safe option, depending on the duration of the job


For two concurrent comparisons, I think it can happen when computing global scores of A and B if two users are to collide at the same time (one can read before the save of the other). For the individual scores, since it is user-based, it should not be an issue.

Since the online heuristic purpose is ultimately to update the global score approximatively, I don't how strict we should be about that error. Hopefully another user compare either A and B and global score should be "corrected" by that error.

A test load should be able to detect if that conflict will happen or the heuristic is fast enough to hold ACID-database properties

Reply:
the heuristic take 8 seconds to complete with public database real data, with 7 seconds for save_tournesol_scores(poll). I'm patching right now to only save tournesol_score of entity A and B.

reapplying the individual scaling take nearly the second left. I'm looking into it to make it faster

Reply:
This total number of entities is used for a single purpose: make the difference between "large" polls (eg "videos") and "small" polls (with very few entities) because they use a slightly different definition for supertrusted users.

As the exact number doesn't really matter, another approach (if necessary) would be to keep this value in cache, to avoid recomputing on each run.